File: src/python/ods/spec/office/document.py
"""
ODF spec element: office:document (ODS root element)

Spec ref: ODF 1.3 §3.1 — Document Root Element
Fact ref: FACT-ODS-EX-0029
QName: office:document
Namespace: urn:oasis:names:tc:opendocument:xmlns:office:1.0
Canonical class: Document
Facade: OdsDocument
"""
from __future__ import annotations
from typing import Any


class Document:
    """Canonical spec-shaped class for office:document in ODS context."""

    spec_qname = "office:document"
    spec_fact_ref = "FACT-ODS-EX-0029"
    namespace_uri = "urn:oasis:names:tc:opendocument:xmlns:office:1.0"
    local_name = "document"
    facade_names = ["OdsDocument"]

    def __init__(self, data: dict[str, Any]) -> None:
        self._data = data

    @property
    def sheet_count(self) -> int:
        return len(self._data.get("sheets", []))

    @property
    def sheets(self) -> list:
        return list(self._data.get("sheets", []))

    @property
    def is_ods(self) -> bool:
        return True

    def to_dict(self) -> dict[str, Any]:
        return dict(self._data)

    def __repr__(self) -> str:
        return f"Document(sheet_count={self.sheet_count})"

File: src/net/fods/Spec/Office/Document.cs
// FormatFactory.Fods — Spec.Office.Document — Canonical spec-shaped model class
// spec_qname: office:document
// spec_fact_ref: FACT-FODS-001
// TC-QHARD-050: converted from architecture_only stub to real model class
namespace FormatFactory.Fods.Spec.Office;

/// <summary>
/// Spec-shaped model class for the ODF office:document element.
///
/// ODF 1.3 §3.1.2 — office:document is the root element of a flat ODF
/// spreadsheet document (single-XML form, .fods).
/// spec_qname: office:document
/// spec_fact_ref: FACT-FODS-001
///
/// This is a canonical class in the Spec/ hierarchy. The facade wrapper is
/// FormatFactory.Fods.FodsDocument (in FodsDocument.cs).
/// </summary>
public sealed class Document
{
    /// <summary>The ODF QName for this element. Grounded in ODF 1.3 §3.1.2.</summary>
    public const string SpecQName = "office:document";

    /// <summary>The SAL fact reference for this element (sal-facts-latest.json).</summary>
    public const string SpecFactRef = "FACT-FODS-001";

    /// <summary>
    /// The office:mimetype attribute value.
    /// For FODS this is "application/vnd.oasis.opendocument.spreadsheet".
    /// </summary>
    public string MimeType { get; init; } =
        "application/vnd.oasis.opendocument.spreadsheet";

    /// <summary>
    /// The office:version attribute value, e.g. "1.3".
    /// </summary>
    public string? Version { get; init; }

    /// <summary>Number of table:table elements (sheets) in this document.</summary>
    public int SheetCount { get; init; }
}
